mapwms.c

```diff
@@ -1079,8 +1075,13 @@ int msWMSLoadGetMapParams(mapObj *map, int nVersion,
       isUsedInNestedGroup = (int*)msSmallCalloc(map->numlayers, sizeof(int));
       msWMSPrepareNestedGroups(map, nVersion, nestedGroups, numNestedGroups, isUsedInNestedGroup);
 
-      for (k=0; k<numlayers; k++) {
-        layerfound = 0;
+      if (ows_request->layerwmsfilterindex != NULL)
+        msFree(ows_request->layerwmsfilterindex);
+      ows_request->layerwmsfilterindex = (int*)msSmallMalloc(map->numlayers * sizeof(int));
```

Comment:
is there a guarantee that all entries of layerwmsfilterindex will be initialized in the following loop ? If not, that might explain the crash you get on some platforms. Perhaps use msSmallCalloc() to make sure that all entries are initialized